components/common/src/command/package/download.rs

```diff
@@ -0,0 +1,400 @@
+//! Downloads a Habitat package from a [depot](../depot).
+//!
+//! # Examples
+//!
+//! ```bash
+//! $ hab pkg download core/redis
+//! ```
+//!
+//! Will download `core/redis` package from a custom depot:
+//!
+//! ```bash
+//! $ hab pkg download core/redis/3.0.1 redis -u http://depot.co:9633
+//! ```
+//!
+//! This would download the `3.0.1` version of redis.
+//!
+//! # Internals
+//!
+//! * Resolve the list of partial artifact identifiers to fully qualified idents
+//! * Gather the TDEPS of the list (done concurrently with the above step)
+//! * Download the artifact
+//! * Verify it is un-altered
+//! * Fetch the signing keys
+
+use std::{collections::HashSet,
+          path::{Path,
+                 PathBuf},
+          time::Duration};
+
+use crate::{api_client::{self,
+                         BoxedClient,
+                         Client,
+                         Error::APIError,
+                         Package},
+            hcore::{self,
+                    crypto::{artifact,
+                             keys::parse_name_with_rev,
+                             SigKeyPair},
+                    fs::{cache_artifact_path,
+                         cache_key_path},
+                    package::{PackageArchive,
+                              PackageIdent,
+                              PackageTarget},
+                    ChannelIdent}};
+
+use reqwest::StatusCode;
+use retry::{delay,
+            retry};
+
+use crate::{error::{Error,
+                    Result},
+            ui::{Status,
+                 UIWriter}};
+
+pub const RETRIES: usize = 5;
+pub const RETRY_WAIT: Duration = Duration::from_millis(3000);
+
+/// Download a Habitat package.
+///
+/// If an `PackageIdentTarget` is given, we retrieve the package from the specified Builder
+/// `url`. Providing a fully-qualified identifer will result in that exact package being installed
+/// (regardless of `channel`). Providing a partially-qualified identifier will result in the
+/// installation of latest appropriate release from the given `channel`.
+///
+/// Any dependencies of will be retrieved from Builder (if they're not already cached locally).
+///
+/// At the end of this function, the specified package and all its dependencies will be downloaded
+/// on the system.
+
+/// Note: it's worth investigating whether
```

Comment:
Nitpicks... fix the comment line lengths to be somewhat similar across the board